delphi/program_analysis/autoTranslate/scripts/translate.py

```diff
@@ -340,35 +341,33 @@ def printAstTree(astFile, tree, blockVal):
     return blockVal
 
 
-def analyze(files, pickleFile, fortranFile):
-    global cleanup
+def get_trees(files: List[str]) -> List:
+    return [ET.parse(f) for f in files]
+
+
+def analyze(trees, comments) -> Dict:
     outputFiles = {}
     ast = []
 
-    # Extracts the comments from the original Fortran source file
-    comments = get_comments(fortranFile)
-
     # Parse through the ast tree once to identify and grab all the funcstions
     # present in the Fortran file.
-    for f in files:
-        tree = ET.parse(f)
+    for tree in trees:
         loadFunction(tree)
 
     # Parse through the ast tree a second time to convert the XML ast format to
     # a format that can be used to generate python statements.
-    for f in files:
-        tree = ET.parse(f)
-        ast += parseTree(tree.getroot(), ParseState())
+    for tree in trees:
+        ast += parseTree(tree, ParseState())
```

Comment:
@skdebray @pratikbhd I changed the following line:

```python
ast += parseTree(tree.getroot(), ParseState())
```

to

```python
ast += parseTree(tree, ParseState())
```

I did this to get the tests working - I noticed that the [`crop_yield.f` example](https://github.com/ml4ai/delphi/blob/program_analysis/delphi/program_analysis/autoTranslate/tests/test_data/crop_yield.f) has changed a bit from the original `crop_yield.f` file that Jon had written. I was not able to process either version with the original version of the line above. However, after changing this line, I was able to process the old version of `crop_yield.f` - which I have now placed [here](https://github.com/ml4ai/delphi/blob/program_analysis/tests/data/crop_yield.f).

My guess is that the `getroot()` method is being called because of some peculiarities with PETPT - is that right? Will this change affect the rest of your set up?

If the `crop_yield.f` file in delphi/tests/data is too outdated, @pratikbhd could you replace it with the updated version and fix the tests as well?